pastas/plotting/bokeh.py

```diff
@@ -0,0 +1,300 @@
+
+    Usage
+    -----
+    >>> ps.extensions.register_bokeh()
+    INFO: Registered bokeh plotting methods in Model class, e.g. `ml.bokeh.plot()`.
+    >>> fig = ml.bokeh.results()
+    >>> fig.write_html("results_figure.html")
+
+    Methods
+    -------
+    plot
+        plot oseries and model simulation, interactive version of `ml.plot()`
+    results
+        plot oseries, model simulation, contribution, step responses and parameters
+        table,interactive version of `ml.plots.results()`
+
+    Notes
+    -----
+    The `bokeh` extension is registered in the `Model` class by calling the
+    `register_bokeh()` function. To work in Juptyer notebooks, the
+    `bokeh.io.output_notebook()` function should be called before plotting. The `bokeh`
+    extension is not registered by default, and should be called explicitly. Check the
+    bokeh documentation for more information on how to interact with the plots.
+
+    """
+
+    def __init__(self, model):
+        self._model = model
+
+    def plot(self, tmin=None, tmax=None, height=300, width=600, show_plot=True):
+        """Plot the observations and model simulation.
+
+        Parameters
+        ----------
+        tmin : pd.Timestamp, optional
+            start time for model simulation, by default None
+        tmax : pd.Timestamp, optional
+            end time for model simulation, by default None
+        height : int, optional
+             height of the plot, by default 500
+        width : int, optional
+           width of the plot, by default 800
+        show_plot : bool, optional
+            Show the plot (i.e., in Jupyter Notebooks), by default True
+
+        Returns
+        -------
+        p : bokeh.plotting.figure
+            Bokeh figure with the observations and model simulation.
+
+        Examples
+        --------
+        >>> ps.extensions.register_bokeh()
+        INFO: Registered bokeh plotting methods in Model class, e.g. `ml.bokeh.plot()`.
+        >>>
+        >>> fig = ml.bokeh.plot()
+
+        """
+
+        data = self._model.get_output_series(tmin=tmin, tmax=tmax, split=False)
+        source = ColumnDataSource(data)
+        rsq = self._model.stats.rsq(tmin=tmin, tmax=tmax)
+
+        TOOLS = "zoom_in,zoom_out,reset,pan,xwheel_zoom,box_zoom,undo"
+
+        p = figure(
+            title="Pastas Model",
+            y_axis_label="Head",
+            x_axis_location=None,
+            tools=TOOLS,
+            width=width,
+            height=height,
+            x_axis_type="datetime",
+            toolbar_location="above",
+        )
+
+        p.scatter(
+            "index",
+            "Head_Calibration",
+            source=source,
+            legend_label="Observations",
+            color="black",
+            alpha=0.7,
+        )
+        p.line(
+            "index",
+            "Simulation",
+            source=source,
+            legend_label=r"Simulation (R$$^2$$ = {:.2f})".format(rsq),
```

Comment:
The superscript isn't formatting nicely for me.

Reply:
Same here, I'll have a look.

Reply:
https://docs.bokeh.org/en/3.4.0/docs/user_guide/styling/mathtext.html Not yet supported for legend_labels. So I removed it.